test/e2e/performanceprofile/functests/12_hypershift/hypershift.go

```diff
@@ -0,0 +1,200 @@
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+		workerRTNodes, err = nodes.GetByLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		hostedClusterName, err := hypershift.GetHostedClusterName()
+		Expect(err).ToNot(HaveOccurred())
+		nodePools, err = ListNodePools(context.TODO(), testclient.ControlPlaneClient, hostedClusterName)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(nodePools).ToNot(BeEmpty(), "no node pools found")
+		Expect(len(nodePools)).To(BeNumerically(">=", 2))
+		PrintNodePoolProfiles(nodePools)
+	})
+
+	Context("Multiple Nodepool", Ordered, func() {
+		isolated := performancev2.CPUSet("1-2")
+		reserved := performancev2.CPUSet("0,3")
+		policy := "best-effort"
+
+		It("should verify support for different performance profiles on hosted cluster via multiple node pools", func() {
+			By("Creating a deep copy of the performance profile for the second node pool")
+			secondProfile = profile.DeepCopy()
+			secondProfile.Name = "second-profile"
+			np = nodePools[1]
+			By("Creating the second performance profile in the control plane")
+			Expect(testclient.ControlPlaneClient.Create(context.TODO(), secondProfile)).To(Succeed(), "Failed to create the performance profile")
+
+			By("Attaching the tuning object to the second node pool")
+			Expect(nodepools.AttachTuningObject(context.TODO(), testclient.ControlPlaneClient, secondProfile, nodePools[1])).To(Succeed())
+
+			By("Waiting for the second node pool configuration to start updating")
+			err = nodepools.WaitForUpdatingConfig(context.TODO(), testclient.ControlPlaneClient, np.Name, np.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for the second node pool configuration to be ready")
+			err = nodepools.WaitForConfigToBeReady(context.TODO(), testclient.ControlPlaneClient, np.Name, np.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking if reserved CPUs are correctly set on the worker nodes")
+			for _, node := range workerRTNodes {
+				By(fmt.Sprintf("Checking kubelet configuration for node %s", node.Name))
+
+				result, err := chkKubeletConfigFn(context.TODO(), &node)
+				Expect(err).ToNot(HaveOccurred(), "Failed to fetch kubelet configuration")
+
+				obj, err := manifestsutil.DeserializeObjectFromData([]byte(result), kubeletconfigv1beta1.AddToScheme)
+				Expect(err).ToNot(HaveOccurred(), "Failed to deserialize kubelet configuration")
+
+				kc, ok := obj.(*kubeletconfigv1beta1.KubeletConfiguration)
+				Expect(ok).To(BeTrue(), "Deserialized object is not of type KubeletConfiguration")
+
+				Expect(kc.ReservedSystemCPUs).To(Equal("0"), "ReservedSystemCPUs is not correctly set")
+				Expect(kc.TopologyManagerPolicy).To(Equal("single-numa-node"), "TopologyManagerPolicy is not correctly set")
+			}
+		})
+
+		It("should verify that Performance Profile update re-creates only target nodepool nodes", func() {
```

Comment:
if we ever want to focus on this specific test, would that be possible considering the main configuration of the second profile is done in the former It()?

Reply:
Moved the profile configuration steps to the Before block to isolate its dependence on the previous It().